Monthly summaries

jose-alatorre opened the pull request:
1. downloaded missing PeMS files + script to uncompress text files
2. Updated [summarize-to-hour-and-timeperiod.R](https://github.com/BayAreaMetro/pems-typical-weekday/blob/monthly-summaries/src/summarize-to-hour-and-timeperiod.R) to create the monthly files needed in addition to the two existing.
3. excluded more holidays when adding in other months: 
4. Updated [combine-all-years-districts.R](https://github.com/BayAreaMetro/pems-typical-weekday/blob/monthly-summaries/src/combine-all-years-districts.R) to aggregate monthly files to a single all-year-aggregate file (in addition to other files it's already aggregating)
5. Saved the all-year-aggregate files to M:\Data\Traffic\PeMS 
6. Committed tableau for visualizing aggregate file to [pems-typical-weekday/notebooks](https://github.com/BayAreaMetro/pems-typical-weekday/tree/monthly-summaries/notebooks)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207182936724120

jose-alatorre committed:
Create extract gz.py

extract txt files that were not historically relevant to existing 2 tableau workbooks that show data at an annual level

jose-alatorre committed:
renamed old file to better match observed naming convention

jose-alatorre committed:
update to the pems-typical-weekday

Updated summarize-to-hour-and-timeperiod.R to create the monthly files needed in addition to the two existing.

Updated combine-all-years-districts.R to aggregate monthly files to a single all-year-aggregate file (in addition to other files it's already aggregating)

jose-alatorre committed:
edited scripts + updated files

to include month column in freeway data
to average by same days/method for bridge data

jose-alatorre committed:
New Tableau workbook to visualize traffic volumes

minor update to bridge traffic script to better integrate with tableau

jose-alatorre committed:
uploading the edited script in this central folder

shortened the script as the daily files are no longer needed

lmz committed:
Renaming PeMS-script.py to download-PeMS-data.py

for clarity

lmz commented:
Thank you for this pull request, @jose-alatorre!  I'll start reviewing and making some small updates to prepare for merge but overall, looking good!

lmz committed:
Expand this script to handle other datasets

e.g.  tmg_trucks_hour, station_day
Also prompts the user for some parameters rather than having them embedded in the script

jose-alatorre committed:
Update Traffic Volume Trends.twb

minor formatting changes + aggregation fix for average daily traffic volumes

jose-alatorre committed:
Merge branch 'monthly-summaries' of https://github.com/BayAreaMetro/pems-typical-weekday into monthly-summaries

Commit:
updated files with a field for length to use as a weight for averages

jose-alatorre committed:
see summary

updated 2022 data through December
updated tableau workbook (last sheet had broken, presumably due to folder location changing)
updated extract_gz.py to include a progress bar
added length field to sum_for_hours_all_months for weighted average calculation

lmz committed:
Merge branch 'master' into monthly-summaries

lmz committed:
Small updates:

mkdir zip_dir if needed
unzip the gzip data file for station_hour files

lmz committed:
Minor improvements (they shouldn't affect resulting output files)

1) Add script documentation
2) Change input parameters to be script arguments
3) Write suspect station files out for later perusal
4) Deprecate write_annual_district_to_disk() since it wasn't really doing anything useful; the code is clearer inline
5) Rename annual_period_all_months_sum_df to be more clear -- this dataframe is by hour (not period) and by month.

lmz committed:
Keep metadata in final files in the same order as before

lmz committed:
Add metadata columns to one more group_by

lmz committed:
Move file loop to within prefix (rather than the other way around)

lmz committed:
Make metadata a requirement

lmz committed:
Print value counts for year to check if any are missing

lmz committed:
Refresh data for recent code changes

These should not really change the data

lmz committed:
Refresh tableau

lmz committed:
Last cosmetic updates
